packages/Search/examples/distributed_tree_driver/distributed_tree_driver.cpp

```diff
@@ -0,0 +1,296 @@
+/****************************************************************************
+ * Copyright (c) 2012-2018 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <DTK_DistributedSearchTree.hpp>
+
+#include <Kokkos_DefaultNode.hpp>
+#include <Teuchos_CommandLineProcessor.hpp>
+#include <Teuchos_DefaultComm.hpp>
+#include <Teuchos_StandardCatchMacros.hpp>
+#include <Teuchos_Time.hpp>
+#include <Teuchos_TimeMonitor.hpp>
+
+#include <cmath> // cbrt
+#include <random>
+
+template <class NO>
+int main_( Teuchos::CommandLineProcessor &clp, int argc, char *argv[] )
+{
+    Teuchos::Time timer( "timer" );
+    Teuchos::TimeMonitor time_monitor( timer );
+
+    using DeviceType = typename NO::device_type;
+    using ExecutionSpace = typename DeviceType::execution_space;
+
+    int n_values = 50000;
+    int n_queries = 20000;
+    int n_neighbors = 10;
+    double overlap = 0.;
+    int dim = 3;
+    bool perform_knn_search = true;
+    bool perform_radius_search = true;
+
+    clp.setOption( "values", &n_values,
+                   "number of indexable values (source) per MPI rank" );
+    clp.setOption( "queries", &n_queries,
+                   "number of queries (target) per MPI rank" );
+    clp.setOption( "neighbors", &n_neighbors,
+                   "desired number of results per query" );
+    clp.setOption( "overlap", &overlap, "overlap of the point clouds" );
+    clp.setOption( "dim", &dim, "dimension of the global point cloud" );
```

Comment:
No, this is misleading, that's not what `dim` is.

Reply:
then what?

Reply:
Point cloud is 3D in this driver regardless of what this variable is.  What you named `dim` guides the relative placement of the MPI rank point clouds or, in some sense, the way we partition the global distributed cloud of points.
* `dim == 1`, clouds are stacked with some offset along the x-axis
* `dim == 2`, clouds are arranged like tiles (?) on the xy-plan
* `dim == 3`, ...

Reply:
> What you named dim guides the relative placement of the MPI rank point clouds or, in some sense, the way we partition the global distributed cloud of points.

Yes, I do understand how the code I wrote works. @aprokop  doesn't like `alignment_dim` so I change to `dim` which you don't like. So which name do you want?

Reply:
So effectively `dim` represents "how many dimensions of space is the problem partitioned in", correct? Effectively, it is determining a partitioning algorithm. How about `partition_algorithm_dimension` or some abbreviation of that?